cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -184,8 +187,6 @@ public KafkaAssemblyOperator(Vertx vertx, boolean isOpenShift,
                 .compose(state -> state.zkHeadlessServiceEndpointReadiness())
                 .compose(state -> state.zkPersistentClaimDeletion())
                 .compose(state -> state.kafkaUpgrade())
-                .compose(state -> state.kafkaManualPodCleaning())
```

Comment:
If you move the `kafkaManualPodCleaning()` and `kafkaManualRollingUpdate()` before `kafkaUpgrade()` then won't the `KafkaCluster` could generate a `StatefulSet` using the upgraded image without going through the upgrade process?

Reply:
The user is annotating pod and not changing the `version` field in the `KafkaCluster` resource and the others needed for upgrading. Why with a manual action did before it should get the new image?

Reply:
> The user is annotating pod and not changing the version field in the KafkaCluster resource and the others needed for upgrading.

You don't know what the user's done since the last reconciliation. They could have changed both, plus a few other things. However, thinking more about it, so long as the STS hasn't changed then I guess this shouldn't be a problem.